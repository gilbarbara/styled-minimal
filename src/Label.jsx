import React from 'react'; //eslint-disable-line no-unused-vars
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { inline } = props;
  const {
    color: labelColor,
    fontWeight: labelFontWeight,
    inlineFontSize,
    marginBottom,
  } = themeGet(props, 'label');

  return css`
    align-items: center;
    color: ${labelColor};
    display: flex;
    font-family: inherit;
    font-weight: ${labelFontWeight};
    ${inline ? `font-size: ${px(inlineFontSize)}` : ''};
    line-height: ${baseStyles.lineHeight};
    ${!inline ? `margin-bottom: ${px(marginBottom)}` : ''};
    white-space: nowrap;
  `;
};

export const Label = styled(Box)`
  ${styles};
`;

Label.displayName = 'Label';

Label.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  inline: PropTypes.bool,
  ...Box.propTypes,
};

Label.defaultProps = {
  as: 'label',
};

export default Label;
