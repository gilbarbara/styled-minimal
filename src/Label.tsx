import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, lineHeightStyles, sx, system } from './utils/system';

import { LabelProps } from './types';

export const StyledLabel = styled('label')<LabelProps>(
  props => {
    const { inline } = props;
    const { color, inlineFontSize, marginBottom } = getTheme(props, 'label');

    return css`
      ${baseStyles};
      align-items: center;
      color: ${color};
      cursor: pointer;
      display: flex;
      font-family: inherit;
      ${inline ? `font-size: ${px(inlineFontSize)}` : ''};
      line-height: ${lineHeightStyles};
      ${!inline ? `margin-bottom: ${px(marginBottom)}` : ''};
      white-space: nowrap;
    `;
  },
  sx,
  system,
);

export const Label = React.forwardRef<HTMLLabelElement, LabelProps>((props, ref) => (
  <StyledLabel ref={ref} {...props} />
));

Label.defaultProps = {
  inline: false,
};
