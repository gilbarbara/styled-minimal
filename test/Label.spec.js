import React from 'react';
import { mount } from 'enzyme';

import Label from '../src/Label';

describe('Label', () => {
  const wrapper = mount(<Label>Far far away</Label>);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
