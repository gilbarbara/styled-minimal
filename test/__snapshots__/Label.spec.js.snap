// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #343434;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: inherit;
  font-weight: bold;
  line-height: 1.4;
  margin-bottom: 8px;
  white-space: nowrap;
}

<Label
  as="label"
>
  <label
    className="c0"
  >
    Far far away
  </label>
</Label>
`;
